tests/unit/models/LifecycleConfiguration.js

```diff
@@ -0,0 +1,271 @@
+const assert = require('assert');
+const { parseString } = require('xml2js');
+
+const LifecycleConfiguration =
+    require('../../../lib/models/LifecycleConfiguration.js');
+
+/**
+ * Format of xml request:
+ <LifecycleConfiguration>
+    <Rule>
+        <ID>id1</ID>
+        <Filter>
+            <Prefix>logs/</Prefix>
+        </Filter>
+        <Status>Enabled</Status>
+        <Expiration>
+            <Days>365</Days>
+        </Expiration>
+    </Rule>
+    <Rule>
+        <ID>DeleteAfterBecomingNonCurrent</ID>
+        <Filter>
+            <And>
+                <Prefix>logs/</Prefix>
+                <Tag>
+                    <Key>key1</Key>
+                    <Value>value1</Value>
+                </Tag>
+            </And>
+        </Filter>
+        <Status>Enabled</Status>
+        <NoncurrentVersionExpiration>
+            <NoncurrentDays>1</NoncurrentDays>
+        </NoncurrentVersionExpiration>
+    </Rule>
+</LifecycleConfiguration>
+ */
+
+const requiredTags = [
+    { tag: 'LifecycleConfiguration',
+        errMessage: 'request xml is undefined or empty' },
```

Comment:
Checking error messages sounds a little bit overkill, compared to the additional cost to maintain the error messages both in the feature code and the tests (to fix typos/reformulate). I would rather check only the error codes (names) which are less likely to change than the full message.

Reply:
I agree, but since so many of the errors returned are `MalformedXML`, I wanted to be sure that the error was being thrown in the correct place.
That was especially true when I was ironing out the logic for the parsing, but if you still think it should be changed, let me know :)

Reply:
No you can leave as is no worry :)